web/packages/teleport/src/Roles/RoleEditor/StandardEditor/sections.tsx

```diff
@@ -66,17 +67,56 @@ export const SectionBox = ({
   validation?: ValidationResult;
   onRemove?(): void;
 }>) => {
+  enum ExpansionState {
+    /** The section is fully collapsed. */
+    Collapsed,
+    /** The section is fully expanded. */
+    Expanded,
+    /**
+     * The section is still expanded, but it's showing a collapsing animation.
+     * In this state, the <details> element is still open to make all its
+     * children visible.
+     */
+    Collapsing,
+  }
+
   const theme = useTheme();
-  const [expanded, setExpanded] = useState(true);
-  const ExpandIcon = expanded ? Minus : Plus;
-  const expandTooltip = expanded ? 'Collapse' : 'Expand';
+  const [expansionState, setExpansionState] = useState(ExpansionState.Expanded);
+  const expandTooltip =
+    expansionState === ExpansionState.Collapsed ? 'Collapse' : 'Expand';
   const validator = useValidation();
+  // Points to the content element whose height will be observed for setting
+  // target height of the expand animation.
+  const contentRef = useRef();
+  const [contentHeight, setContentHeight] = useState(0);
+
+  useResizeObserver(
+    contentRef,
+    entry => {
+      setContentHeight(entry.borderBoxSize[0].blockSize);
+    },
+    { enabled: true }
```

Comment:
We can remove this parameter, it's enabled by default.

Reply:
That argument is actually mandatory, the only option would be to pass an empty object, so i decided to be more explicit just for readability's sake. I'd go ahead and modify it where it's declared, but I was not sure if I'd get into conflicts with Maxim, who also intended to modify something there.